contracts/contracts/SimpleWriteAccessController/library.cairo

```diff
@@ -0,0 +1,136 @@
+%lang starknet
+
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.bool import TRUE, FALSE
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+
+from ownable import Ownable_only_owner, Ownable_initializer
+
+@event
+func AddedAccess(user : felt):
+end
+
+@event
+func RemovedAccess(user : felt):
+end
+
+@event
+func CheckAccessEnabled():
+end
+
+@event
+func CheckAccessDisabled():
+end
+
+@storage_var
+func s_check_enabled() -> (checkEnabled : felt):
+end
+
+@storage_var
+func s_access_list(address : felt) -> (bool : felt):
+end
+
+# Adds an address to the access list
+@external
+func add_access{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(user : felt):
+    Ownable_only_owner()
+    _add_access(user)
+    return ()
+end
+
+func _add_access{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(user : felt):
+    let (has_access) = s_access_list.read(user)
+    if has_access == FALSE:
+        s_access_list.write(user, TRUE)
+        AddedAccess.emit(user)
+        return ()
+    end
+
+    return ()
+end
+
+# # Removes an address from the access list
+@external
+func remove_access{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(user : felt):
+    Ownable_only_owner()
+    let (has_access) = s_access_list.read(user)
+    if has_access == TRUE:
+        s_access_list.write(user, FALSE)
+        RemovedAccess.emit(user)
+        return ()
+    end
+
+    return ()
+end
+
+# makes the access check enforced
+@external
+func enable_access_check{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}():
+    Ownable_only_owner()
+    let (check_enabled) = s_check_enabled.read()
+    if check_enabled == FALSE:
```

Comment:
do we need to explicitly do `== FALSE`?  Can we just do

```
if( !s_check_enabled.read())

```

Reply:
No